cpp/inc/bond/core/apply.h

```diff
@@ -16,26 +16,26 @@ namespace detail
 {
 
 /// @brief Apply transform to serialized struct wrapped in bonded<T>
-template <typename Transform, typename T, typename Reader>
+template <typename Protocols, typename Transform, typename T, typename Reader>
```

Comment:
Any motivation for having Protocols as first template parameter rather than, say, last?